src/Symfony/Component/Notifier/Bridge/LinkedIn/LinkedInTransport.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\LinkedIn;
+
+use Symfony\Component\Notifier\Bridge\LinkedIn\Share\AuthorShare;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Component\Notifier\Exception\TransportException;
+
+/**
+ * @author Smaïne Milianni <contact@smaine.me>
+ *
+ * @experimental in 5.2
+ *
+ * @see https://docs.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/ugc-post-api#sharecontent
+ */
+final class LinkedInTransport extends AbstractTransport
+{
+    protected const PROTOCOL_VERSION = '2.0.0';
+    protected const PROTOCOL_HEADER = 'X-Restli-Protocol-Version';
+    protected const ENDPOINT = 'https://api.linkedin.com/v2/ugcPosts';
+
+    private $authToken;
+    private $accountId;
+
+    public function __construct(string $authToken, string $accountId, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->authToken = $authToken;
+        $this->accountId = $accountId;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('linkedin:://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof LinkedInOptions);
+    }
+
+    /**
+     * @see https://docs.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/ugc-post-api
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, ChatMessage::class, \get_class($message)));
+        }
+
+        if ((null !== $options = $message->getOptions()) && !$message->getOptions() instanceof LinkedInOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, LinkedInOptions::class));
+        }
+
+        if (!($opts = $message->getOptions()) && $notification = $message->getNotification()) {
```

Comment:
`$options` instead of `$message->getOptions()` or remove this variable (`$options`).

Reply:
why ?

Reply:
Sorry, this is my mistake.